### PR TITLE
consolidate PHP eval scripts

### PR DESCRIPTION
**Pull request by ahebrank**
Addresses #23 

Whether to sandbox or not is currently determined by $sandbox_me at the the top of the script: https://github.com/ahebrank/phpepl/compare/mrjoelkemp:master...master#diff-610eaebfa315a3e4829a75c4459b6b82R3

This PR also has a Vagrantfile for folks like me with that dev pattern.

### CODE REVIEW

**Comment by ahebrank** [comment]
Travis is failing because of https://github.com/calvinmetcalf/diffie-hellman/issues/5 today.  Maybe it'll work tomorrow :)

**Comment by mrjoelkemp** [comment]
Wow. This is great work @ahebrank! Thanks for contributing.

Odd about that dependency fail. I subscribed to the thread and will rerun travis when the problem has been fixed. 

In the mean time, I'll make sure all is well locally.

File: src/eval/index.php [hunk 1]
```diff
@@ -1,9 +1,12 @@
 <?php
+	// set me to false to turn off sandboxing
+	$sandbox_me = true;
```
Comment (mrjoelkemp) [inline src/eval/index.php]: This forces users who want to locally serve the repl to modify the code. It's not a huge modification, but the environment variable solution avoids this. Was there a particular reason why you went the boolean route? Just curious.

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# set up a Vagrantbox to run the repl
+# check it out at http://phpepl.dev after running 'vagrant up'
```
Comment (mrjoelkemp) [inline Vagrantfile]: I'd like to add some of these details to the readme so users can take advantage of this awesome setup. Is it as simple as having vagrant installed, cd'ing into the phpepl root and then running vagrant up?

File: Vagrantfile [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  # Remove the default Vagrant directory sync
+  config.vm.synced_folder ".", "/vagrant", disabled: true
+
+  config.vm.synced_folder "./", "/var/www/public", id: "vagrant-root",
+    owner: "vagrant",
+    group: "www-data",
+    mount_options: ["dmode=775,fmode=664"]
+
+  config.vm.network :private_network, ip: "192.168.56.101"
```
Comment (mrjoelkemp) [inline Vagrantfile]: I'm familiar with using vagrant, but not its set up. Is there any significance to this IP address? Would it ever need to be changed by a user?

File: src/eval/index.php (hunk 1 above)
Comment (ahebrank) [inline src/eval/index.php]: Simplicity, mostly. Environmental variables will also vary by host (if running via Vagrant, for instance, localhost won't work as an IP check), and it's good to always default to sandboxed and force the user to change if they need to.

File: Vagrantfile (hunk 2 above)
Comment (ahebrank) [inline Vagrantfile]: Pretty much.  Also need to have virtualbox installed.

File: Vagrantfile (hunk 3 above)
Comment (ahebrank) [inline Vagrantfile]: No, it seems to be a common one.  If a user doesn't have the vagrant-hostsupdate plugin installed, the local address will be that address.

File: Vagrantfile (hunk 2 above)
Comment (ahebrank) [inline Vagrantfile]: Also `vagrant plugin install vagrant-hostsupdater` to get the hostname automatically in /etc/hosts or whatever the host equivalent.

File: src/eval/index.php (hunk 1 above)
Comment (mrjoelkemp) [inline src/eval/index.php]: Thanks for clarifying. For user ease, I was thinking:

```php
  $sandbox_me = isset($_ENV['DEP_NAME']);
```

That env var (name is at the mercy of cloudcontrol, unfortunately) wouldn't need to be set by users (unless they want to force local sandboxing). That var would be set for all production hosting environments though (cloudcontrol and heroku).  

This way, a user simply has to download the project, serve locally, and then visit the local hostname for an unsandboxed experience – no fiddling with the code is necessary. It also makes for a seamless upgrade path.

I'll merge this in as is and, if you agree, change it to the env var solution in a follow up commit.

**Comment by mrjoelkemp** [comment]
Merged in via cee7abc2f17fd7fa3edfa3522d8c46728a1bc0c4. Thanks again!

File: src/eval/index.php (hunk 1 above)
Comment (ahebrank) [inline src/eval/index.php]: Sure, that works.  My theoretical worry is that someone would put up an unsandboxed version on a vhost without realizing, but you're probably right that people who run locally likely want an unsandboxed version without need to edit source.

Comment (mrjoelkemp) [inline src/eval/index.php]: Done and pushed live. Works well. Thanks again for the help.

**Pull request closed without merging**